src/vs/workbench/services/workingCopy/common/workingCopyService.ts

```diff
@@ -138,6 +139,11 @@ export class WorkingCopyService extends Disposable implements IWorkingCopyServic
 	registerWorkingCopy(workingCopy: IWorkingCopy): IDisposable {
 		const disposables = new DisposableStore();
 
+		// {{SQL CARBON EDIT}} @chlafreniere need to block working copies of notebook editors from being tracked
+		if (workingCopy.resource.path.includes('notebook-editor-') && workingCopy.resource.scheme === Schemas.untitled) {
+			return disposables;
```

Comment:
How does this impact the saving/hot exit functionality? It looks like this class tracks dirty state so is that going to affect whether it prompts to save when you try to close a dirty notebook?

Reply:
Saving works as expected now. was able to replace an unsaved with a saved nb editor without issue.

For dirty state, I can see the dot. No prompts when closing ADS for unsaved notebooks.